src/writer.rs

```diff
@@ -296,7 +300,11 @@ impl FdtWriter {
 
     /// Write a property containing an array of 64-bit unsigned integers.
     pub fn property_array_u64(&mut self, name: &str, cells: &[u64]) -> Result<()> {
-        let mut arr = Vec::with_capacity(cells.len() * size_of::<u64>());
+        let size = cells
+            .len()
+            .checked_mul(size_of::<u64>())
```

Comment:
Same concern as `property_array_u32` - may be clearer to just use `size_of_val` rather than re-multiplying it.